### PR TITLE
feat: Add support to pass in mouseHandlers into IrisGrid

### PR DESCRIPTION
**Pull request by mofojed**
- Move the doubleClick behaviour from IrisGridDataSelectMouseHandler into GridSelectionMouseHandler, so it's baked into Grid itself (double-clicking a grid in the styleguide was exhibiting erroneous behaviour)
- Allow passing in mouseHandlers to IrisGrid. Will be necessary to support https://github.com/deephaven/deephaven-plugins/issues/320
- Tested using my branch for `on_row_press` support on deephaven.ui: https://github.com/deephaven/deephaven-plugins/pull/346
- Verified Linker still worked as expected

### CODE REVIEW

File: packages/iris-grid/src/mousehandlers/IrisGridDataSelectMouseHandler.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ import type IrisGrid from '../IrisGrid';
  */
 class IrisGridDataSelectMouseHandler extends GridMouseHandler {
   constructor(irisGrid: IrisGrid) {
-    super();
+    super(825);
```
Comment (bmingles) [inline packages/iris-grid/src/mousehandlers/IrisGridDataSelectMouseHandler.ts:15]: This could probably use a comment as to what `825` is

Comment (bmingles) [reply packages/iris-grid/src/mousehandlers/IrisGridDataSelectMouseHandler.ts:15]: ah, I see it is `order` now in the base class.

**Review by bmingles — CHANGES_REQUESTED** [review]
Left a question about event handlers

File: packages/iris-grid/src/mousehandlers/IrisGridDataSelectMouseHandler.ts (hunk 1 above)
Comment (mofojed) [reply packages/iris-grid/src/mousehandlers/IrisGridDataSelectMouseHandler.ts:15]: Yea, we could (and probably should) have constants for a few different levels of this... https://github.com/deephaven/web-client-ui/issues/1859

File: packages/iris-grid/src/IrisGrid.tsx [hunk 2]
```diff
@@ -1394,6 +1403,26 @@ class IrisGrid extends Component<IrisGridProps, IrisGridState> {
...
+
+  getKeyHandlers(): readonly KeyHandler[] {
+    const { keyHandlers } = this.props;
+    return this.getCachedKeyHandlers(keyHandlers);
+  }
+
+  getCachedMouseHandlers = memoize(
+    (
+      mouseHandlers: readonly GridMouseHandler[]
+    ): readonly GridMouseHandler[] => [...mouseHandlers, ...this.mouseHandlers]
```
[the remark this answers is not in the document]
Comment (mofojed) [reply packages/iris-grid/src/IrisGrid.tsx:1418]: `this.mouseHandlers` and `this.keyHandlers` are not getting by initial props, that's a local value that is initialized in the constructor.
Then the `getCachedMouseHandlers` combines the default/code IrisGrid mouseHandlers with the custom provided `mouseHandlers` provided from the props.
We use a similar pattern in `Grid.tsx` as well.

[the remark this answers is not in the document]
Comment (bmingles) [reply packages/iris-grid/src/IrisGrid.tsx:1418]: ah, misread. Looks good then

**Review by bmingles — DISMISSED** [review]
LGTM

**Review by bmingles — APPROVED** [review]
LGTM

**Pull request merged**